cmd/misc/main.go

```diff
@@ -1233,3 +1240,294 @@ OUTER:
+			return
+		}
+		if lastEpoch > 0 { // guard against underflows
+			lastEpoch = lastEpoch - 1
+		}
+	} else {
+		_, lastEpoch = utils.GetFirstAndLastEpochForDay(endDay)
+	}
+
+	start := time.Now()
+
+	epochsPerDay := utils.EpochsPerDay()
+	if lastEpoch < epochsPerDay {
+		logrus.Infof("skipping exporting stats, first day has not been indexed yet")
+		return
+	}
+	currentDay := lastEpoch / epochsPerDay
+	previousDay := currentDay - 1
+
+	for day := startDay; day <= previousDay; day++ {
+		startDay := time.Now()
+		err := UpdateValidatorStatisticsSyncData(day, rpcClient, dryRun)
+		if err != nil {
+			utils.LogError(err, fmt.Errorf("error exporting stats for day %v", day), 0)
+			break
+		}
+
+		logrus.Infof("finished updating validators_stats for day %v, took %v", day, time.Since(startDay))
+	}
+
+	logrus.Infof("finished all exporting stats for days %v - %v, took %v", startDay, previousDay, time.Since(start))
+}
+
+func UpdateValidatorStatisticsSyncData(day uint64, client rpc.Client, dryRun bool) error {
+	exportStart := time.Now()
+	firstEpoch, lastEpoch := utils.GetFirstAndLastEpochForDay(day)
+
+	logrus.Infof("exporting statistics for day %v (epoch %v to %v)", day, firstEpoch, lastEpoch)
+
+	if err := db.CheckIfDayIsFinalized(day); err != nil && !dryRun {
+		return err
+	}
+
+	logrus.Infof("getting exported state for day %v", day)
+
+	var err error
+	maxValidatorIndex := uint64(1999999) //bt.GetMaxValidatorindexForEpoch(lastEpoch)
+	// if err != nil {
+	// 	logrus.Errorf("error getting max validator index for epoch %v: %v", lastEpoch, err)
+	// 	return err
+	// }
+	validators := make([]uint64, 0, maxValidatorIndex)
+	validatorData := make([]*types.ValidatorStatsTableDbRow, 0, maxValidatorIndex)
+	validatorDataMux := &sync.Mutex{}
+
+	logrus.Infof("processing statistics for validators 0-%d", maxValidatorIndex)
+	for i := uint64(0); i <= maxValidatorIndex; i++ {
+		validators = append(validators, i)
+		validatorData = append(validatorData, &types.ValidatorStatsTableDbRow{
+			ValidatorIndex: i,
+			Day:            int64(day),
+		})
+	}
+
+	g := &errgroup.Group{}
+
+	g.Go(func() error {
+		if err := db.GatherValidatorSyncDutiesForDay(validators, day, validatorData, validatorDataMux); err != nil {
+			return fmt.Errorf("error in GatherValidatorSyncDutiesForDay: %w", err)
+		}
+		return nil
+	})
+
+	var statisticsData1d []*types.ValidatorStatsTableDbRow
+	g.Go(func() error {
+		var err error
+		statisticsData1d, err = db.GatherStatisticsForDay(int64(day) - 1) // convert to int64 to avoid underflows
+		if err != nil {
+			return fmt.Errorf("error in GatherPreviousDayStatisticsData: %w", err)
+		}
+		return nil
+	})
+
+	err = g.Wait()
+	if err != nil {
+		return err
+	}
+
+	onlySyncValidatorData := make([]*types.ValidatorStatsTableDbRow, 0, len(validatorData))
+	for index := range validatorData {
+
+		if validatorData[index].ParticipatedSync > 0 || validatorData[index].MissedSync > 0 || validatorData[index].OrphanedSync > 0 {
```

Comment:
You have to check the total columns too, these validators & their respective total columns have to get updated as well.

Reply:
This is a filter to sync out all validators that had sync committees that day. Total will be updated further below, but if the validator was not in a sync committee that day total will stay the same anyway.

Reply:
Let's assume you use this tool over a day range that includes 2 distinct sync committee periods, with no validators being in both periods:
Day `x`: validators in sync period `y` have sync duties => their sync stats are updated; sync period `y+1` starts
Day `x+1`: validators in sync period `y` have **no** sync duties => their sync stats are not updated, even though their total changed the previous day.
Pls correct me if I'm wrong.

Reply:
Am not sure what you are getting at. On Day `x` sync stats are updated including the total. So the total stats on `x+1` for the previous day are already correct?